external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriter.scala

```diff
@@ -94,4 +94,10 @@ private[kafka010] object KafkaWriter extends Logging {
       }
     }
   }
+
+  def close(sc: SparkContext, kafkaParams: ju.Map[String, Object]): Unit = {
+    sc.parallelize(1 to 10000).foreachPartition { iter =>
+      CachedKafkaProducer.close(kafkaParams)
+    }
```

Comment:
This would cause  `CachedKafkaProducer.close` to be executed on each executor. I am thinking of a better way here.